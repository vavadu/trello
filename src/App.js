import React, {useEffect} from 'react';
import useAuth from "./data/hooks/useAuth";
import {Routes, Route, useNavigate} from "react-router-dom";
import Login from './Pages/Login/Login';
import { Register } from "./Pages/Register";
import Main from "./Pages/Board/Board";


function App() {
    const { authUser, isLoadingUser } = useAuth();
    let navigate = useNavigate();

    useEffect(() => {
        authUser ? navigate('/') : navigate('/login');
    }, [authUser])

    return (
        <>
            <Routes>
                <Route path="/" element={<Main/>}/>
                <Route path="/login" element={<Login/>}/>
                <Route path="/register" element={<Register/>}/>
            </Routes>
        </>


        )



    // Add some React loader to app
    // if (isLoadingUser) {
    //     return <Loader />;
    // }



}

export default App;