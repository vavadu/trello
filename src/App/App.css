.App {
    width: 100%;
    display: flex;
}

.Main-section__wrapper {
    padding: 50px 90px;
}

.Cards__wrapper {
    display: flex;
    flex-wrap: nowrap;
}
