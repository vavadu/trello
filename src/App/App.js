import React from "react";
import Aside from "../Aside/Aside";
import Header from "../Header/Header"
import Card from "../Card/Card";
import './App.css';

function App() {
    return (
        <div className="App">
            <Aside />
            <div className="Main-section__wrapper">
                <Header />
                <div className="Cards__wrapper">
                    <Card title="To Do"/>
                    <Card title="In progress"/>
                    <Card title="Testing"/>
                    <Card title="Done"/>
                </div>
            </div>
        </div>
    );
}

export default App;
