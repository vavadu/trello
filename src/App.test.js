import React from 'react';
import { shallow } from 'enzyme';
import App from './App.js';
import { QueryClientProvider, QueryClient } from 'react-query';

describe("testing App", () => {

    // test("should match snapshot", () => {
    //     const queryClient = new QueryClient();
    //     const wrapper = shallow(
    //         <QueryClientProvider client={queryClient}>
    //             <App />
    //         </QueryClientProvider>
    //     );
    //     expect(wrapper.html()).toMatchSnapshot();
    // })
})
