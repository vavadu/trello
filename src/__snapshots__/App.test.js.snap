// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App should match snapshot 1`] = `"<div class=\\"spinner\\"><div class=\\"blob top\\"></div><div class=\\"blob bottom\\"></div><div class=\\"blob left\\"></div><div class=\\"blob move-blob\\"></div></div>"`;
